cmake/Hardening.cmake

```diff
@@ -0,0 +1,205 @@
+#Anyone is free to copy, modify, publish, use, compile, sell, or distribute this software, either in source code form or as a compiled binary, for any purpose, commercial or non-commercial, and by any means.
+#In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain. We make this dedication for the benefit of the public at large and to the detriment of our heirs and successors. We intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.
+#THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#For more information, please refer to <https://unlicense.org/>
+
+include(CheckCXXCompilerFlag)
+
+option(HARDENING_SSE2 "Enable hardening flags requiring at least SSE2 support for target" OFF)
+
+function(determineSupportedHardeningFlags property)
+	set(FLAGS_HARDENING "")
+	foreach(flag ${ARGN})
+		unset(var_name)
+		string(REPLACE "=" "_eq_" var_name ${flag})
+		string(REPLACE "," "_comma_" var_name ${var_name})
+		set(var_name "SUPPORTS_HARDENING_${property}_${var_name}")
+		check_cxx_compiler_flag(${flag} ${var_name})#since linker flags and other flags are in the form of compiler flags
+		if(${${var_name}})
+			list(APPEND FLAGS_HARDENING "${flag}")
+		endif()
+	endforeach(flag)
+	string(REPLACE ";" " " FLAGS_HARDENING "${FLAGS_HARDENING}")
+	#message(STATUS "FLAGS_HARDENING ${FLAGS_HARDENING}")
+	set(HARDENING_${property} "${FLAGS_HARDENING}" CACHE STRING "Hardening flags")
+endfunction(determineSupportedHardeningFlags)
+
+function(processFlagsList target property)
+	get_target_property(FLAGS_UNHARDENED ${target} ${property})
+	if(FLAGS_UNHARDENED MATCHES "FLAGS_UNHARDENED-NOTFOUND")
+		set(FLAGS_UNHARDENED "")
+	endif()
+	#message(STATUS "processFlagsList ${target} ${property} ${FLAGS_UNHARDENED}")
+	#message(STATUS "HARDENING_${property} ${HARDENING_${property}}")
+	if(HARDENING_${property})
+	else()
+		determineSupportedHardeningFlags(${property} ${ARGN})
+	endif()
+	
+	set(FLAGS_HARDENED ${FLAGS_UNHARDENED})
+	list(APPEND FLAGS_HARDENED ${HARDENING_${property}})
+	string(REPLACE ";" " " FLAGS_HARDENED "${FLAGS_HARDENED}")
+	#message(STATUS "${target} PROPERTIES ${property} ${FLAGS_HARDENED}")
+	set_target_properties(${target} PROPERTIES ${property} "${FLAGS_HARDENED}")
+endfunction(processFlagsList)
+
+function(setupPIC target)
+	set_property(TARGET ${target} PROPERTY POSITION_INDEPENDENT_CODE ON) # FUCK, doesn't work
```

Comment:
Is that profanity really necessary here?

Reply:
No, I just copied that file from my tool.